lib/maze/bitbar_devices.rb

```diff
@@ -8,7 +9,56 @@ class BitBarDevices
     APPIUM_1_15_0 = '1.15.0'
     APPIUM_1_20_2 = '1.20.2'
 
+    BASE_URI = 'https://cloud.bitbar.com/api/v2/me'
+    FILTER_PATH = 'devices/filters'
+
+    DEVICE_GROUP_IDS = {
+      # Classic, non-specific devices for each Android version
+      'ANDROID_10_0' => '46024',
+
+      # iOS devices
+      'IOS_14' => '46025'
+    }
+
     class << self
+      def call_bitbar_api(path, query, api_key)
+        encoded_query = URI.encode_www_form(query)
+        uri = URI("#{BASE_URI}/#{path}?#{encoded_query}")
+        request = Net::HTTP::Get.new(uri)
+        request.basic_auth(api_key, '')
+
+        res = Net::HTTP.start(uri.hostname, uri.port, use_ssl: true) do |http|
+          http.request(request)
+        end
+
+        JSON.parse(res.body)
+      end
+
+      def get_filtered_device_name(device_group_id, api_key)
+        path = "device-groups/#{device_group_id}/devices"
+        query = {
+          'filter': "online_eq_true"
+        }
+        all_devices = call_bitbar_api(path, query, api_key)
+        filtered_devices = all_devices['data'].reject { |device| device['locked'] }
+        filtered_devices.first['displayName']
+      end
+
+      def get_device(device_group, platform, platform_version, api_key)
+        device_group_id = DEVICE_GROUP_IDS[device_group]
+        device_name = get_filtered_device_name(device_group_id, api_key)
+        device_caps = case platform.downcase
```

Comment:
Rubocop says this assignment before return is redundant and that `device_caps` can just be removed.  I'm not a great fan of Ruby's "magic" return behaviour, but that's not a battle I'm going to win and so I'm prepared to be more-or-less a slave to Rubocop.